lib/devise/controllers/helpers.rb

```diff
@@ -46,30 +55,30 @@ def authenticate_#{group_name}!(favorite = nil, opts = {})
                 mappings.unshift mappings.delete(favorite.to_sym) if favorite
                 mappings.each do |mapping|
                   opts[:scope] = mapping
-                  warden.authenticate!(opts) if !devise_controller? || opts.delete(:force)
+                  warden.authenticate!(*self.class.strategies_for(mapping), opts) if !devise_controller? || opts.delete(:force)
```

Comment:
You'll likely also want to respect any per-controller strategies here (and add `self.class.strategies_for` to this `.authenticate` call):

https://github.com/heartcombo/devise/blob/main/app/controllers/devise/sessions_controller.rb#L19

Reply:
I guess that `Devise::SessionsController` probably doesn't include these controller helpers, but as part of this work, it may be worth considering how to allow custom strategies in that controller. (It would be a mildly more complete solution, even if it's just adding an empty `auth_strategies` or `strategies` method to that controller and respecting it in the `authenticate` call.)

Reply:
it is now possible to override the SessionsController too :)